custom/src/main/java/co/elastic/otel/dynamicconfig/DynamicConfiguration.java

```diff
@@ -131,6 +134,44 @@ public void stopDisablingAllTraces() {
     reenableTracesFor(ALL_INSTRUMENTATION);
   }
 
+  // okay to synchronize as this should only be called after multi-second intervals and
+  // additionally only called from threads which are not doing anything application blocking
+  public synchronized void deactivateInstrumentations(String deactivateList) {
+    if (deactivateList != null && !deactivateList.trim().isEmpty()) {
+      // some values in the disable_instrumentations list
+      Set<String> toBeEnabled = null;
```

Comment:
[minor] making this non-nullable would remove a couple of null-checks below

Reply:
the null status is a specific state where there is something already deactivated. If there is nothing deactivated the path is different. Essentially making this non-null would then require another boolean variable

Reply:
not relevant with the changed flow